meta-balena-common/recipes-containers/docker-disk/files/entry.sh

```diff
@@ -65,7 +65,12 @@ done
 if [ -n "${TARGET_REPOSITORY}" ] && [ -n "${TARGET_TAG}" ]; then
 	_supervisor_image=$(balena_api_fetch_image_from_app "${TARGET_REPOSITORY}" "${TARGET_TAG#v}" "${BALENA_API_ENV}" "${BALENA_API_TOKEN}")
 	echo "Pulling ${TARGET_REPOSITORY}:${TARGET_TAG}"
-	docker pull "${_supervisor_image}"
+	if docker pull "${_supervisor_image}"; then
+		docker tag "${_supervisor_image}" "${_supervisor_image%@*}"
```

Comment:
@klutchell I have a PR to do something similar but with a more user-friendly name in https://github.com/balena-os/meta-balena/pull/2473/commits/0cf7765cf038b45d5b90adbb82e58b8d5f8c0ccb#diff-4f634ac771e1a812dc0ad46d0fe6dd51756f31ba10300af95e6a6375ed732287R170

```
$DOCKER tag "${image_id}" "balena_supervisor:${version}"
```

However, the part where you say this avoid redownloads I need to understand. Do you mind if I look into this as part of testing the PR I mentioned above?

Reply:
@alexgg I'll take a look at your PR, but the current issue is we are preloading the SV by pulling `repo@digest` so anywhere that we refer to just `SUPERVISOR_IMAGE` which is just the `repo` the image will not be found.

```
docker pull repo@digest
docker run/inspect repo <-- object not found
docker run/inspect repo@digest <-- works
```
So the current releases are pulling the supervisor image on first boot, because the run and inspect commands fail to find the image. My detailed findings are in #2492